File: JavaScript_Basic_04_2023_SolutionsOfProblems/05.Ex_PB-JS-While-Loop/P04.Walking.js
function walking(input) {
    let daySteps = 0;
    let i = 0;
    while (daySteps < 10000) {
        if (input[i] === "Going home") {
            daySteps += Number(input[i + 1]);
            break;
        }
        daySteps += Number(input[i]);
        i++;
    }
    if (daySteps < 10000) {
        console.log(`${10000 - daySteps} more steps to reach goal.`);
    } else {
        console.log("Goal reached! Good job!");
        console.log(`${daySteps - 10000} steps over the goal!`);
    }
}
walking(["1500",
    "3000",
    "250",
    "1548",
    "2000",
    "Going home",
    "2000"])

File: Java_Basic_Jan_2023_SolutionsOfProblems/P05Ex_While_Loop/P04_Walking.java
package P05Ex_While_Loop;

import java.util.Scanner;

public class P04_Walking {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        String input = "";
        int sumSteps = 0;
        while (sumSteps < 10000) {
            input = scanner.nextLine();
            if (input.equals("Going home")) {
                input = scanner.nextLine();
                sumSteps += Integer.parseInt(input);
                break;
            }
            sumSteps += Integer.parseInt(input);
        }
        if (sumSteps < 10000) {
            System.out.printf("%d more steps to reach goal.", 10000 - sumSteps);
        } else {
            System.out.printf("Goal reached! Good job!%n" + "%d steps over the goal!", sumSteps - 10000);
        }
    }
}
